### PR TITLE
[CLI] Adds Testing to CLI

### PR DESCRIPTION
**Pull request by TomMcHugh**
This PR adds the necessary testing to the install command for the cli. 

Testing checks for: Podfile existing, Containing React-Native Tags, and for React Pod to be included.

### CODE REVIEW

**Comment by frantic** [comment]
Why did you close this? I don't know if we have this in the docs, but to run the tests do

```shell
$ npm test
```

we use Jest for testing, but unfortunately it doesn't work on `node` version above 0.10. I use [n](https://www.npmjs.com/package/n) to switch node binaries, but it's up to you.

Also I believe in order for Jest to find your test it has to be named `moduleYouTesting-test.js` (at least that's our convention)

File: local-cli/install.js [hunk 1]
```diff
@@ -83,6 +83,20 @@ module.exports = {
         throw e;
       }
     }
+
+    return {
+      podfileExists: function() {
```
Comment (frantic) [inline local-cli/install.js]: [fs.existsSync](https://nodejs.org/api/fs.html#fs_fs_existssync_path)

**Comment by TomMcHugh** [comment]
@frantic Ahh that makes sense, some of it was failing in Travis CI, but working when doing npm test, so I'll figure that out and commit.

**Comment by frantic** [comment]
@TomMcHugh I think the problem is in using current dir in both tests and code. `cwd()` is super unreliable and often leads to weird bugs. Prefer passing around absolute paths and using `path.resolve` and friends with `__dirname`, `__filename`.

For example, your `install` function can take `root` as param. In your tests, you can create fake files to simulate different layouts (with or without Podfile, with or without node_modules, etc.). But `cli.js` will  invoke `install` function with real root dir (either by passing `cwd()` or `__dirname`, etc.)

**Comment by TomMcHugh** [comment]
@frantic Yep that looks like it. All is well with Travis.

**Comment by brentvatne** [comment]
@frantic - is this still relevant? seems like a clean merge still if not :smile:

**Comment by sahrens** [comment]
We have a nice end-to-end test, so not sure if still relevant - @frantic: can you either close out or review if you want it?

**Comment by sahrens** [comment]
Actually Tadeu's been looking at this stuff more lately and frantic is on vacation...

**Comment by tadeuzagallo** [comment]
yaytests

Tests are always relevant! :smile:
Just re-triggering travis since a *lot* has changed in the mean time.

**Pull request merged**